src/trivial.rs

```diff
@@ -0,0 +1,301 @@
+//! Trivial transmutation, content-wise
+//!
+//! Functions in this module are guarded from out-of-bounds memory access and
+//! from unsafe transmutation target types through the use of the
+//! [`TriviallyTransmutable`](trait.TriviallyTransmutable.html)) trait.
+//! If a certain type can be safely constructed out of any byte combination,
+//! then it may implement this trait. This is the case for primitive integer
+//! types (e.g. `i32`, `u32`, `i64`), arrays of other trivially transmutable
+//! types, and `repr(C)` structs composed.
+//!
+//! However, they are still not entirely safe because the source data may not
+//! be correctly aligned for reading and writing a value of the target type.
+//! The effects of this range from less performance (e.g. x86) to trapping or
+//! address flooring (e.g. ARM), but this is undefined behavior nonetheless.
+
+
+use self::super::guard::{PermissiveGuard, PedanticGuard, Guard};
+use self::super::base::{transmute_many, from_bytes};
+#[cfg(feature = "std")]
+use self::super::base::transmute_vec;
+use self::super::Error;
+
+
+/// Type that can be constructed from any combination of bytes.
```

Comment:
maybe slap "can be non-`unsafe`ly transmuted into" in parens at the end here

Reply:
That would be inaccurate, though. It can still be unsafe due to memory alignment. I would rather not do that and defer to the *Safety* section of the trait.

Reply:
valid concern!